### ISSUE TITLE
Call to throwing function gets compiler error with `try!`

### ISSUE THREAD

**Issue by daviditen** [chapel-lang/chapel#13274]
Calls to throwing functions get compiler errors even when using the `try!` operator at expression level.

```chapel
module M {
  var f = try! open("trybang.chpl", iomode.r);
  var ch = try! f.reader();
}
```

```bash
tryExprOpen.chpl:2: error: call to throwing function open_default_hints without throws, try, or try! (relaxed mode)
$CHPL_HOME/modules/standard/IO.chpl:1517: note: throwing function open_default_hints defined here
tryExprOpen.chpl:2: error: call to throwing function open_default_style without throws, try, or try! (relaxed mode)
$CHPL_HOME/modules/standard/IO.chpl:1518: note: throwing function open_default_style defined here
tryExprOpen.chpl:2: error: call to throwing function open_default_url without throws, try, or try! (relaxed mode)
$CHPL_HOME/modules/standard/IO.chpl:1518: note: throwing function open_default_url defined here
tryExprOpen.chpl:3: error: call to throwing function reader_default_hints without throws, try, or try! (relaxed mode)
$CHPL_HOME/modules/standard/IO.chpl:2448: note: throwing function reader_default_hints defined here
tryExprOpen.chpl:3: error: call to throwing function reader_default_style without throws, try, or try! (relaxed mode)
$CHPL_HOME/modules/standard/IO.chpl:2449: note: throwing function reader_default_style defined here
```

```bash
% chpl --version
chpl version 1.20.0 pre-release (837a25cb3a)
Copyright (c) 2004-2019, Cray Inc.  (See LICENSE file for more details)
```

**Comment by daviditen** [comment]
Moving the `try!` out to statement level compiles:

```chapel
module M {
  try! {
    var f = open("trybang.chpl", iomode.r);
    var ch = f.reader();
  }
}
```

**Comment by daviditen** [comment]
A more standalone version:

```chapel
module M {
  config const cond = false;

  proc thrower() throws {
    if cond then
      throw new owned Error();
    return 1;
  }

  proc myproc(arg = thrower()) throws {
    writeln(arg);
    return arg;
  }

  var a = try! myproc();
  writeln(a);
}
```

A function marked `throws` with a default argument that is a throwing function call causes the error.  If `myproc` is not marked `throws` this code compiles.

**Comment by daviditen** [comment]
Fixed with #13437.
